src/status_im/ui/screens/contacts/subs.cljs

```diff
@@ -101,7 +102,8 @@
 (reg-sub :contact
   (fn [db]
     (let [identity (:contacts/identity db)]
-      (get-in db [:contacts/contacts identity]))))
+      (or (get-in db [:contacts/contacts identity])
+          (assoc (events/whisper-id->new-contact identity) :new? true)))))
```

Comment:
Basically here we generate the contact if it's not in the list already and we let the UI know adding a new? flag in the profile.
Otherwise we could just return `nil` from the subscription and let the UI generate the user. Not sure what's  best, I would like to keep the UI without much logic, and seems like generating a profile there might not be the best, what do you reckon?

Reply:
better to generate in view, because this will be recalculated on any change in app-db